BaseCollector.py

```diff
@@ -162,3 +167,96 @@ def wait_for_inventory_data(self):
                 print("waiting for initial iteration: " + type(self).__name__)
         print("done: initial query " + type(self).__name__)
         return
+
+    def generate_gauges(self, metric_type, calling_class, vrops_entity_name, labelnames):
+        if not isinstance(labelnames, list):
+            print("Can't generate Gauges without label list, called from", calling_class)
+            return False
+        # switching between metric and property types
+        if metric_type == 'metric':
+            statkey_yaml = self.read_collector_config()['statkeys']
+            gauges = dict()
+            for statkey_pair in statkey_yaml[calling_class]:
+                statkey_suffix = statkey_pair['metric_suffix']
+                gauges[statkey_suffix] = {
+                    'gauge': GaugeMetricFamily('vrops_' + vrops_entity_name + '_' + statkey_suffix.lower(),
+                                               'vrops-exporter', labels=labelnames),
+                    'statkey': statkey_pair['statkey']
+                }
+            return gauges
+
+        if metric_type == 'property':
+            properties_yaml = self.read_collector_config()['properties']
+            if 'number_metrics' in properties_yaml[calling_class]:
+                gauges = dict()
+                for property_pair in properties_yaml[calling_class]['number_metrics']:
+                    property_suffix = property_pair['metric_suffix']
+                    gauges[property_suffix] = {
+                        'gauge': GaugeMetricFamily('vrops_' + vrops_entity_name + '_' + property_suffix.lower(),
+                                                   'vrops-exporter', labels=labelnames),
+                        'property': property_pair['property']
+                    }
+                return gauges
+            else:
+                return {}
+        else:
+            return False
```

Comment:
This is never evaluated in code. If it really hits this return value it would error in `do_metric` the first time the expected dict is looped. `gauges` then would be bool, instead of list. Maybe add some instance check `do_metrics` or check if it evaluates to something but False.

Reply:
Yes, the return is now also changed to a empty dict. This makes it more general usable in terms of adding new properties and statkeys to `collector_config`.